app/src/main/java/com/github/fribourgsdp/radio/AddPlaylistActivity.kt

```diff
@@ -3,31 +3,57 @@ package com.github.fribourgsdp.radio
 import android.content.Intent
 import android.os.Bundle
 import android.view.View
-import android.widget.*
+import android.widget.ArrayAdapter
+import android.widget.EditText
+import android.widget.Spinner
+import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
-import androidx.core.view.get
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
 
 class AddPlaylistActivity : AppCompatActivity() {
     private val listSongs = ArrayList<Song>()
     private var listNames = ArrayList<String>()
-    private lateinit var listAdapter: ArrayAdapter<String>
-    private lateinit var listView : ListView
+    private lateinit var listAdapter: SongAdapter
+    private lateinit var recyclerView : RecyclerView
     private lateinit var errorTextView : TextView
     private lateinit var genreSpinner: Spinner
     lateinit var user : User
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_add_playlist)
-        listAdapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, listNames)
-        listView = findViewById(android.R.id.list)
-        listView.adapter = listAdapter
+
         genreSpinner = findViewById(R.id.genreSpinner)
         genreSpinner.adapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, Genre.values())
         errorTextView = findViewById(R.id.addPlaylistErrorTextView)
 
         user = UserProfileActivity.loadUser(this)
+
+        processIntent(intent)
+
+        recyclerView = findViewById(R.id.list_playlist_creation)
+        listAdapter = SongAdapter(listSongs, object : OnClickListener{
+            override fun onItemClick(position: Int) {
+                listSongs.removeAt(position)
+                listAdapter.notifyItemRemoved(position)
+            }
+        })
+
+        recyclerView.adapter = listAdapter
+        val layoutManager = LinearLayoutManager(this)
+        layoutManager.stackFromEnd = true
+        recyclerView.layoutManager = layoutManager
+        errorTextView = findViewById(R.id.addPlaylistErrorTextView)
+
```

Comment:
Ideally this would use the recycler view fragment but I agree not using it here because of the specifics of format it takes. Possibly a later refactor

Reply:
Exactly, I had the same idea